### PR TITLE
[PN-272] Modifying Azure Python dependency to v2

### PR DESCRIPTION
**Pull request by kenazk**
Fixes issue in PN-272:

image

### CODE REVIEW

**Comment by kenazk** [comment]
Do we have a convention in our `step.sh` for providing status updates to stdout? Would be nice to have some more log messages. Something like: 

```
[INFO] Authenticating to Azure CLI...
```

**Comment by kenazk** [comment]
I'm also trying to add the template outputs to Nebula outputs. I could use some help from one of you bash ninjas: 

```
az group deployment show -n nebula-aks-deployment -g kenaz142 | jq '.properties.outputs | to_entries '
[
  {
    "key": "controlPlaneFQDN",
    "value": {
      "type": "String",
      "value": "nebula-1352-3ac189c7.hcp.westus2.azmk8s.io"
    }
  },
  {
    "key": "controlPlaneFQDN2",
    "value": {
      "type": "String",
      "value": "nebula-1352-3ac189c7.hcp.westus2.azmk8s.io"
    }
  },
  {
    "key": "controlPlaneFQDN3",
    "value": {
      "type": "String",
      "value": "nebula-1352-3ac189c7.hcp.westus2.azmk8s.io"
    }
  }
]
```
What's a concise way of setting the Nebula outputs with these values? Otherwise, it's going to be all PM glue+stick code :)

**Comment by kenazk** [comment]
Took a pass at adding outputs. It's super janky I know.

File: azureresourcemanager-deployer/step.sh [hunk 1]
```diff
@@ -87,3 +90,25 @@ else
...
+else
+  OUTPUTS=$(az deployment show -n ${DEPLOYMENT_NAME} | $JQ -r '.properties.outputs | to_entries')
+fi
+
+length=$( $JQ '. | length' <<< $OUTPUTS )
+if [ ${length} -gt 0 ]
+then
+  echo "[INFO] Setting Nebula outputs ..."
+  KEYS=( $( $JQ -r ' {key: .[].key} | .["key"]' <<< $OUTPUTS))
+  VALUES=( $( $JQ -r ' {value: .[].value.value} | .["value"]' <<< $OUTPUTS))
```
Comment (kenazk) [inline azureresourcemanager-deployer/step.sh]: This is not going to work for `object` outputs. In my case, returning an array of base64 encoded kubeconfigs.

Comment (kenazk) [reply azureresourcemanager-deployer/step.sh]: Need something like this 

```
az group deployment show -n ${DEPLOYMENT_NAME} -g ${RESOURCE_GROUP} | jq -r '.properties.outputs.clustercredentials.value.kubeconfigs | .[].value' | base64 --decode
```

File: azureresourcemanager-deployer/step.sh [hunk 2]
```diff
@@ -41,6 +41,7 @@ if [ -n "${CERT}" ] ; then
 fi
 [ -z "${PASSWORD_OR_CERT}" ] && usage 'spec: your `azure` hash must specify either a `password` or a `cert` key'
 
+echo "[INFO] Logging in to Azure CLI ..."
```
Comment (impl) [inline azureresourcemanager-deployer/step.sh]: The command you're looking for is `ni log info "your message here"` :-).

**Comment by kenazk** [comment]
Ok sounds good. Will close this one.

**Pull request closed without merging**